ush/forecast_predet.sh

```diff
@@ -0,0 +1,299 @@
+  CASE=${CASE:-C768}
+  CDATE=${CDATE:-2017032500}
+  DATA=${DATA:-$pwd/fv3tmp$$}    # temporary running directory
+  ROTDIR=${ROTDIR:-$pwd}         # rotating archive directory
+  ICSDIR=${ICSDIR:-$pwd}         # cold start initial conditions
+}
+
+DATM_predet(){
+  SYEAR=$(echo  $CDATE | cut -c1-4)
+  SMONTH=$(echo $CDATE | cut -c5-6)
+  SDAY=$(echo   $CDATE | cut -c7-8)
+  SHOUR=$(echo  $CDATE | cut -c9-10)
+  # directory set up
+  if [ ! -d $DATA ]; then mkdir -p $DATA; fi
+  if [ ! -d $DATA/DATM_INPUT ]; then mkdir -p $DATA/DATM_INPUT; fi
+  FHMAX=${FHMAX:-9}
+  # Go to Run Directory (DATA)         
+  cd $DATA
+}
+
+FV3_GFS_predet(){
+  echo "SUB ${FUNCNAME[0]}: Defining variables for FV3GFS"
+  CDUMP=${CDUMP:-gdas}
+  CDUMPwave="${CDUMP}wave"
+  FHMIN=${FHMIN:-0}
+  FHMAX=${FHMAX:-9}
+  FHOUT=${FHOUT:-3}
+  FHZER=${FHZER:-6}
+  FHCYC=${FHCYC:-24}
+  FHMAX_HF=${FHMAX_HF:-0}
+  FHOUT_HF=${FHOUT_HF:-1}
+  NSOUT=${NSOUT:-"-1"}
+  FDIAG=$FHOUT
+  if [ $FHMAX_HF -gt 0 -a $FHOUT_HF -gt 0 ]; then FDIAG=$FHOUT_HF; fi
+  WRITE_DOPOST=${WRITE_DOPOST:-".false."}
+  restart_interval=${restart_interval:-0}
+  rst_invt1=`echo $restart_interval |cut -d " " -f 1`
+  
+  PDY=$(echo $CDATE | cut -c1-8)
+  cyc=$(echo $CDATE | cut -c9-10)
+  
+  # Directories.
+  pwd=$(pwd)
+  NWPROD=${NWPROD:-${NWROOT:-$pwd}}
+  HOMEgfs=${HOMEgfs:-$NWPROD}
+  FIX_DIR=${FIX_DIR:-$HOMEgfs/fix}
+  FIX_AM=${FIX_AM:-$FIX_DIR/fix_am}
+  export FIX_AER=${FIX_AER:-$FIX_DIR/fix_aer}
+  export FIX_LUT=${FIX_LUT:-$FIX_DIR/fix_lut}
+  FIXfv3=${FIXfv3:-$FIX_DIR/fix_fv3_gmted2010}
+  DATA=${DATA:-$pwd/fv3tmp$$}    # temporary running directory
+  ROTDIR=${ROTDIR:-$pwd}         # rotating archive directory
+  ICSDIR=${ICSDIR:-$pwd}         # cold start initial conditions
+  DMPDIR=${DMPDIR:-$pwd}         # global dumps for seaice, snow and sst analysis
+  
+  # Model resolution specific parameters
+  DELTIM=${DELTIM:-225}
+  layout_x=${layout_x:-8}
+  layout_y=${layout_y:-16}
+  LEVS=${LEVS:-65}
+  
+  # Utilities
+  NCP=${NCP:-"/bin/cp -p"}
+  NLN=${NLN:-"/bin/ln -sf"}
+  NMV=${NMV:-"/bin/mv"}
+  SEND=${SEND:-"YES"}   #move final result to rotating directory
+  ERRSCRIPT=${ERRSCRIPT:-'eval [[ $err = 0 ]]'}
+  KEEPDATA=${KEEPDATA:-"NO"}
+  
+  # Other options
+  MEMBER=${MEMBER:-"-1"} # -1: control, 0: ensemble mean, >0: ensemble member $MEMBER
+  ENS_NUM=${ENS_NUM:-1}  # Single executable runs multiple members (e.g. GEFS)
+  PREFIX_ATMINC=${PREFIX_ATMINC:-""} # allow ensemble to use recentered increment
+  
+  # IAU options
+  DOIAU=${DOIAU:-"NO"}
+  IAUFHRS=${IAUFHRS:-0}
+  IAU_DELTHRS=${IAU_DELTHRS:-0}
+  IAU_OFFSET=${IAU_OFFSET:-0}
+  
+  # Model specific stuff
+  FCSTEXECDIR=${FCSTEXECDIR:-$HOMEgfs/sorc/fv3gfs.fd/NEMS/exe}
+  FCSTEXEC=${FCSTEXEC:-fv3_gfs.x}
+  PARM_FV3DIAG=${PARM_FV3DIAG:-$HOMEgfs/parm/parm_fv3diag}
+  PARM_POST=${PARM_POST:-$HOMEgfs/parm/post}
+  
+  # Wave coupling parameter defaults to false
+  cplwav=${cplwav:-.false.}
```

Comment:
This is new compared with f/c-c, cplwav defaulting to false is fine, I'm just curious why this was needed here though?  I mean there's not cplflx=false, etc. (That I saw)

Reply:
This is from the develop branch. My speculation is, the config files in f/c-c always take care of the cpl switches including this one, while the config files in develop sometimes don't. So they need a default value here in the script. @KateFriedman-NOAA @WalterKolczynski-NOAA Is this line about cplwav default value actually required in the develop branch? Can we remove it?

Reply:
At the very least, this is probably the wrong location for it. It's definitely set in the config file now so it should be able to be removed.

Reply:
Resolved